src/OpenTelemetry.Instrumentation.StartupHook/StartupHook.cs

```diff
@@ -65,4 +78,46 @@ private static string GetLoaderAssemblyLocation()
             throw;
         }
     }
+
+    private static string GetApplicationName()
+    {
+        var applicationName = string.Empty;
+        try
+        {
+            applicationName = AppDomain.CurrentDomain.FriendlyName;
+        }
+        catch (Exception ex)
+        {
+            StartupHookEventSource.Log.Error($"Error getting AppDomain.CurrentDomain.FriendlyName: {ex}");
+        }
+
+        return applicationName;
+    }
+
+    private static bool IsApplicationInExcludeList(string applicationName)
+    {
+        return GetExcludedApplicationNames().Contains(applicationName);
+    }
+
+    private static IList<string> GetExcludedApplicationNames()
+    {
+        var excludedProcesses = new List<string>();
+
+        var environmentValue = Environment.GetEnvironmentVariable("OTEL_PROFILER_EXCLUDE_PROCESSES");
+
+        if (environmentValue == null)
+        {
+            return excludedProcesses;
+        }
+
+        foreach (var processName in environmentValue.Split(','))
+        {
+            if (!string.IsNullOrWhiteSpace(processName))
+            {
+                excludedProcesses.Add(processName.Trim());
+            }
```

Comment:
We can deal with whitespaces only once

```suggestion
            processName = processName.Trim();
            if (!string.IsNullOrEmpty(processName))
            {  
                excludedProcesses.Add(processName);
            }
```

Reply:
In this case, if you perform the `.Trim()` call before the null check, you can trigger an exception. The code in this method has the same logic as its native counterpart where the profiler checks the same environment variable, so that the code handles extra whitespace more consistently.